server/sockets.js

```diff
@@ -96,6 +97,25 @@ function socketConnection(socket) {
 			return;
 		}
 
+		if (message.data.type === "doc") {
+			getBoard(boardName).then(boardData => {
+				let existingDocuments = 0;
+				for (key in boardData.board) {
+					if (boardData.board[key].type === "doc") {
+						existingDocuments += 1;
+					}
+					if (existingDocuments >= config.MAX_DOCUMENTS) {
+						console.warn("Received too many documents");
+						return;
+					}
+				}
+			});
+
+			if (!message.data.size || message.data.size > config.MAX_DOCUMENT_SIZE) {
```

Comment:
`.length`, not `.size`, no ? This would have been a security issue. It may be a good idea to add a test in `integration.js` to test that large documents are indeed refused.

Reply:
Size is a property of the message but I agree that it should be replaced by `message.data.data.length` to actually combat a malicious message